docs/_theme/djangodocs/layout.html

```diff
@@ -24,7 +24,7 @@
 
 {% block linktags %}
 {{ super() }}
-  <link rel="canonical" href="https://hasura.io/docs/1.0/{{ pagename }}{{ file_suffix }}" />
+  <link rel="canonical" href="https://{{ BASE_DOMAIN }}/docs/1.0/{{ pagename }}{{ file_suffix }}" />
```

Comment:
Actually the canonical can still be hasura.io, hardcoded, irrespective of whether it is staging or production.